chia/wallet/puzzles/p2_singleton.clvm

```diff
@@ -0,0 +1,61 @@
+(mod (SINGLETON_MOD_HASH SINGLETON_MOD_HASH_HASH SINGLETON_ID singleton_innerpuzhash my_id)
+
```

Comment:
Need comments! What does each parameter represent? What's being curried into the puzzle?

I'm assuming `SINGLETON_MOD_HASH` is the hash of this mod and `SINGLETON_MOD_HASH_HASH` is the hash of the hash, `SINGLETON_ID` is the genesis parent coin id, and these are curried into the singleton permanently.

I'm then assuming that `singleton_innerpuzhash` is just some inner puzzle hash (why not call it `inner_puzzle_hash`?) and `my_id` is... I'm not sure? An identifier of... something?

Reply:
It's not the mod of this puzzle, it's the mod of singleton_top_layer so we can reconstruct the ID of the singleton we're committed to paying to

I've added comments clarifying this